tests/test_json.py

```diff
@@ -170,6 +170,29 @@ class Config:
     assert m.json() == '{"x": "P0DT0H2M3.000000S"}'
 
 
+def test_con_decimal_encode() -> None:
+    """
+    Makes sure a decimal with decimal_places = 0, as well as one with places
+    can handle a encode/decode roundtrip.
+    """
+
+    class Id(ConstrainedDecimal):
+        max_digits = 22
+        decimal_places = 0
+        ge = 0
+
+    ObjId = Id
```

Comment:
This is unnecssary, better to use `condecimal` anyway.

Reply:
You mean the ConstrainedDecimal subclass?  I'm cursion to hear why is condecimal better?

I use the subclasses since typers doesn't like the `id: condecimal(args)` approach, and it's neat with reusable types.